collect_app/src/test/java/org/odk/collect/android/utilities/TextUtilsTest.java

```diff
@@ -0,0 +1,33 @@
+package org.odk.collect.android.utilities;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+
+@RunWith(RobolectricTestRunner.class)
+public class TextUtilsTest {
+
+    /**
+     * Should return null if provided with null and not throw a NPE.
+     */
+    @Test
+    public void textToHtml_BounceNullInput() {
+        String input = null;
+        CharSequence observed = TextUtils.textToHtml(input);
+        assertNull(observed);
+    }
+
+    /**
+     * Should do something not-null-ish if provided with non-null string.
+     */
+    @Test
+    public void textToHtml_CallsHTMLOnRealInput() {
+        String input = "This *bold* markdown is \n- nice,\n- efficient.\n";
+        CharSequence result = TextUtils.textToHtml(input);
+        assertNotNull(result);
```

Comment:
Thanks @srsudar I think that's a very sensible change. I wanted to do something like that but wasn't sure what the best approach might be. Maybe if @batkinson could confirm I'll update the PR for merge?

Re: gradle configs: maybe we can address this by only running one flavour in CI for collect. I'd imagine there is a way to set that as a build parameter in circle.yml, which I've opened a new ticket for adding (#205) once we're set on this PR.